src/main/kotlin/xyz/atrius/waystones/commands/WarpKeyCmd.kt

```diff
@@ -0,0 +1,39 @@
+package xyz.atrius.waystones.commands
+
+import org.bukkit.command.Command
+import org.bukkit.command.CommandSender
+import org.bukkit.entity.Player
+import xyz.atrius.waystones.utility.defaultWarpKey
+import org.bukkit.Bukkit
+
+fun warpKeyCmd (
+    sender: CommandSender,
+    cmd: Command,
+    lbl: String,
+    args: List<String>
+): Boolean {
+    // Permission Check
+    if (!sender.hasPermission("waystones.warpkey")) {
+        sender.sendMessage("§d[Waystones]§r §cYou don't have permission to run this command§r")
+        return true
```

Comment:
In most cases, I'd agree, but in this case the command didn't fail, the command worked as intended, the player just didn't have the correct permission.
It informs the player of that and returns true only to avoid the command usage message cluttering the chat.

Reply:
The idea behind the boolean return value is whether or not the command did what the player wanted or not though. If you asked the command to give you a key but it responded with "no" then that should result in a command failure.